### PR TITLE
[Feature Store] Base Redis default path on `config.feature_store` template

### PR DESCRIPTION
**Pull request by alxtkr77**
https://iguazio.atlassian.net/browse/ML-6068

### CODE REVIEW

File: mlrun/datastore/targets.py [hunk 1]
```diff
@@ -1390,6 +1390,39 @@ class RedisNoSqlTarget(NoSqlBaseTarget):
...
+                    == mlrun.common.schemas.ObjectKind.feature_set
+                    else "vectors"
+                )
+                kind = self.kind
+                name = self._resource.metadata.name
+                project = (
+                    self._resource.metadata.project or mlrun.mlconf.default_project
+                )
+                data_prefix = get_default_prefix_for_target(kind).format(
+                    ds_profile_name=parsed_url.netloc,
```
Comment (gtopper) [inline mlrun/datastore/targets.py:1411]: It doesn't exist in config though.

File: mlrun/datastore/targets.py [hunk 2]
```diff
@@ -1390,6 +1390,39 @@ class RedisNoSqlTarget(NoSqlBaseTarget):
     support_spark = True
     writer_step_name = "RedisNoSqlTarget"
 
+    @property
+    def _target_path_object(self):
+        path = self.path or mlrun.mlconf.redis.url
```
Comment (gtopper) [inline mlrun/datastore/targets.py]:
```suggestion
        url = self.path or mlrun.mlconf.redis.url
```

File: mlrun/datastore/targets.py [hunk 3]
```diff
@@ -1390,6 +1390,39 @@ class RedisNoSqlTarget(NoSqlBaseTarget):
...
+                )
+                data_prefix = get_default_prefix_for_target(kind).format(
+                    ds_profile_name=parsed_url.netloc,
+                    authority=parsed_url.netloc,
+                    project=project,
+                    kind=kind,
+                    name=name,
+                )
+                if path.startswith("rediss://"):
+                    data_prefix.replace("redis://", "rediss://", 1)
```
Comment (gtopper) [inline mlrun/datastore/targets.py]:
```suggestion
                    data_prefix = data_prefix.replace("redis://", "rediss://", 1)
```

File: mlrun/datastore/targets.py (hunk 1 above)
Comment (alxtkr77) [reply mlrun/datastore/targets.py:1411]: Accidently removed from the config. Fixed in https://github.com/mlrun/mlrun/pull/5458

File: mlrun/datastore/targets.py (hunk 3 above)
Comment (alxtkr77) [reply mlrun/datastore/targets.py]: Nice catch!

File: mlrun/datastore/targets.py [hunk 4]
```diff
@@ -1390,6 +1390,39 @@ class RedisNoSqlTarget(NoSqlBaseTarget):
...
+                    authority=parsed_url.netloc,
+                    project=project,
+                    kind=kind,
+                    name=name,
+                )
+                if url.startswith("rediss://"):
+                    data_prefix = data_prefix.replace("redis://", "rediss://", 1)
+                if not self.run_id:
+                    version = self._resource.metadata.tag
+                    name = f"{name}-{version or 'latest'}"
```
Comment (gtopper) [inline mlrun/datastore/targets.py]:
```suggestion
                    version = self._resource.metadata.tag or "latest"
                    name = f"{name}-{version}"
```

File: mlrun/datastore/targets.py [hunk 5]
```diff
@@ -1390,6 +1390,39 @@ class RedisNoSqlTarget(NoSqlBaseTarget):
...
+        if self._resource and url:
+            parsed_url = urlparse(url)
+            if not parsed_url.path or parsed_url.path == "/":
+                kind_prefix = (
+                    "sets"
+                    if self._resource.kind
+                    == mlrun.common.schemas.ObjectKind.feature_set
+                    else "vectors"
+                )
+                kind = self.kind
```
Comment (gtopper) [inline mlrun/datastore/targets.py:1405]: This could be inlined.

**Review by gtopper — APPROVED** [review]
LGTM

**Pull request merged**